app/bundles/PageBundle/EventListener/BuilderSubscriber.php

```diff
@@ -405,8 +418,27 @@ public function onPageDisplay(Events\PageDisplayEvent $event)
                 $savePrefs = $this->renderSavePrefs($params);
                 $content   = str_ireplace(self::saveprefsRegex, $savePrefs, $content);
             }
-        }
 
+            $dom = new DOMDocument('1.0', 'utf-8');
```

Comment:
Can you check if the content contains a preference center slot before creating DOMDocument object and loading HTML in it? It's an expensive operation.